src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +57,366 @@ uint32_t SystemRequest::index = 0;
 const std::string kSYNC = "SYNC";
 const std::string kIVSU = "IVSU";
 
+#define QUERY_APPS_VALIDATION_FAILED ":QUERY_APPS_VALIDATION_FAILED:"
+
+struct comparer {
+ public:
+  bool operator()(const utils::custom_string::CustomString x,
+                  const utils::custom_string::CustomString y) {
+    return x.AsMBString().compare(y.AsMBString()) < 0;
+  }
+};
+
+template <typename ObjectType>
+class QueryAppsDataValidatorTemplate {
+ public:
+  typedef std::set<utils::custom_string::CustomString, comparer>
+      synonyms_set_type;
+  typedef std::map<std::string, synonyms_set_type> synonyms_map_type;
+
+  QueryAppsDataValidatorTemplate(const ObjectType& o,
```

Comment:
@okoreniak We have to avoid using such unclear names like o. But in this particular case may be it's normal. I'm not sure.